### PR TITLE
fix(scaffold): validate + format module name from user

### PR DESCRIPTION
**Pull request by asanehisa**
Fixes https://github.com/yext/pages/issues/505

### CODE REVIEW

File: packages/pages/src/scaffold/modules/generate.ts [hunk 1]
```diff
@@ -92,7 +98,11 @@ const validateModuleName = (
   if (fs.existsSync(modulePath)) {
     return false;
   }
-  return /^[a-zA-Z]+$/.test(moduleName.charAt(0));
+  return (
```
Comment (mkilpatrick) [inline packages/pages/src/scaffold/modules/generate.ts:101]: Does this happen after we set it?

File: packages/pages/src/scaffold/modules/templates.ts [hunk 2]
```diff
@@ -5,8 +5,9 @@ export const moduleCode = (
   useTailwind: boolean
 ): string => {
   const tailwind = useTailwind ? ` className="tailwind"` : ``;
-  const formattedModuleName =
-    moduleName.charAt(0).toUpperCase() + moduleName.slice(1);
+  const formattedModuleName = moduleName
```
Comment (mkilpatrick) [inline packages/pages/src/scaffold/modules/templates.ts:8]: Move the formatting into a function and add tests for it.

File: packages/pages/src/scaffold/modules/generate.ts (hunk 1 above)
Comment (asanehisa) [reply packages/pages/src/scaffold/modules/generate.ts:101]: checks validation while user is trying to input via prompts. prompts library rejects user input if it doesn't pass

File: packages/pages/src/scaffold/modules/templates.ts [hunk 3]
```diff
@@ -69,10 +70,9 @@ export const indexCssCode = (useTailwind: boolean): string => {
   return useTailwind
     ? `.tailwind {
   @tailwind base;
+  @tailwind components;
```
Comment (mkilpatrick) [inline packages/pages/src/scaffold/modules/templates.ts:73]: Your other change shouldn't be here if you merged

Comment (asanehisa) [reply packages/pages/src/scaffold/modules/templates.ts]: ah just merged

File: packages/pages/src/scaffold/modules/generate.test.ts [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+import { validateModuleNameFormat } from "./generate.js";
+
+describe("validateModuleNameFormat", () => {
+  const testCases = [
+    { moduleName: "foo123", expected: true },
+    { moduleName: "foo-bar123", expected: true },
+    { moduleName: "foo_bar", expected: true },
+    { moduleName: "foo_bar", expected: true },
+    { moduleName: "foo$bar", expected: true },
+    { moduleName: "foobar-_$", expected: true },
```
Comment (mkilpatrick) [inline packages/pages/src/scaffold/modules/generate.test.ts:11]: What does this get turned into?

Comment (asanehisa) [reply packages/pages/src/scaffold/modules/generate.test.ts:11]: foobar-_$ gets turned into Foobar_$ for the exported Module const.

File: packages/pages/src/scaffold/modules/templates.ts (hunk 2 above)
Comment (asanehisa) [reply packages/pages/src/scaffold/modules/templates.ts:8]: Oops done

**Review by mkilpatrick — APPROVED** [review]

**Pull request merged**